File: algorithms/Data/Stack/stack_simple_b_parentheless.py
from stack import Stack

def parChecker(symbols):
    s = Stack()
    balanced = True
    index = 0
    while index < len(symbols) and balanced:
        symbol = symbols[index]
        if symbol == "(":
            s.push(symbol)
        else:
            if s.isEmpty():
                balanced = False
            else:
                s.pop()

        index = index + 1

    if balanced and s.isEmpty():
        return True
    else:
        return False

print(parChecker('((()))'))
print(parChecker('(()'))

File: algorithms/Data/Stack/stack_simple_b_parentheless.js
"use strict"
class Stack {
    constructor() {
        this.items = new Array()
    }

    isEmpty() {
        return this.items.length == 0
    }

    push(item) {
        return this.items.push(item)
    }

    pop() {
        return this.items.pop()
    }

    size() {
        return this.items.length
    }
}

var parChecker = function (symbols) {
    var s = new Stack()
    var balanced = true
    var i = 0
    while (i < symbols.length) {
        var symbol = symbols[i]
        if (symbol == "(") {
            s.push(symbol)
        } else {
            if (s.isEmpty()) {
                balanced = false
            } else {
                s.pop()
            }
        }
        i++
    }
    if (balanced && s.isEmpty()) {
        return true
    } else {
        return false
    }
}

console.log(parChecker("(())"))
console.log(parChecker("(()"))